### DISCUSSION TITLE
Use kcert with "kong" ingress controller

### DISCUSSION THREAD

**Discussion by ahmedadell94** [nabsul/kcert#85] [General]
We're currently trying to use kcert to generate certificates to the kubernetes ingresses. We use kong ingress controller (not nginx) in the kubernetes cluster so I guess the ingress challenges will not work.
I've checked the kcert image and found that it uses "nginx" implicitly as shown below:
  "ChallengeIngress": {
    "UseClassName": true,
    "ClassName": "**nginx**",
    "UseAnnotations": false,
    "Annotations": {
      "kubernetes.io/ingress.class": "nginx"
    },
    "UseLabels": false,
    "Labels": null
  } 

Do you have any recommendation as a work around?
Is it doable to add ENV in the image and value for it in the helm chart so that we can change ingress controller?

Thanks in Advance,
Ahmed Adel

**Comment by nabsul** [comment]
Hi Ahmed, all values in the `appsettings.json` file can be modified with environment variables. For example, to change `ChallengeIngress:ClassName`, you would need to set an environment variable called `CHALLENGEINGRESS__CLASSNAME` (with two underscores between the two words).

For now, you should be able to manually edit your deployment to add the environment variable. I have created an issue to add support for environment variables to the Helm chart;  https://github.com/nabsul/kcert/issues/86

**Comment by nabsul** [comment]
Note: If you would like to contribute a pull request for this, please let me know and I'll wait for your submission. Otherwise, I'll work on it when I have some time, hopefully in the next few weeks.

**Comment by ahmedadell94** [comment]
Hi Nabeel,
Thanks for your support.
I've tried to patch the deployment and adding ENV CHALLENGEINGRESS__CLASSNAME to kong and it worked fine, thanks.

I can update the chart accordingly.

**Reply by ahmedadell94** [reply]
Can you grant me permission to be able to make pull request?

Kindly find what will be added:

1. in "values.yaml" :
            challengeIngress:
              className: nginx

2. in templates/070-Deployment.yaml:
            - name: CHALLENGEINGRESS__CLASSNAME
              # Challenge Ingress className (default is nginx)
              value: {{ required "challengeIngress.className is required" .Values.challengeIngress.className}}
